src/Raven.Client/Documents/Session/Operations/QueryOperation.cs

```diff
@@ -182,6 +193,32 @@ internal static T Deserialize<T>(string id, BlittableJsonReaderObject document,
             return result;
         }
 
+        private static ConcurrentDictionary<Type, Type> _wrapperTypes;
+        private const string DummyPropertyName = "Result";
+
+        private static T DeserializeInnerArray<T>(BlittableJsonReaderObject document, string fieldToFetch, InMemoryDocumentSessionOperations session,
+            BlittableJsonReaderArray blittableArray)
+        {
+            document.Modifications = new DynamicJsonValue(document)
+            {
+                [DummyPropertyName] = blittableArray
```

Comment:
Can it happen that `document`  will have `"Result"` (`DummyPropertyName`) property already defined?

Reply:
I mean, will it work:

```
select { o.Lines, Result = o.Company } 
```

and won't cause property name clash ?